x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { CspmRulesStats } from './types';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+
+interface DocCount {
+  doc_count: number;
+}
+
+interface BenchmarkName {
+  metrics: { 'rule.benchmark.name': string };
+}
+
+interface BenchmarkId {
+  metrics: { 'rule.benchmark.id': string };
+}
+
+interface BenchmarkVersion {
+  metrics: { 'rule.benchmark.version': string };
+}
+interface RuleName {
+  metrics: { 'rule.name': string };
+}
+interface RuleSection {
+  metrics: { 'rule.section': string };
+}
+interface RuleVersion {
+  metrics: { 'rule.version': string };
+}
+interface RuleNumber {
+  metrics: { 'rule.benchmark.rule_number': string };
+}
+interface PostureType {
+  metrics: { 'rule.benchmark.posture_type': string };
+}
+
+interface RulesStats {
+  rules: {
+    buckets: RuleEntity[];
+  };
+}
+
+interface RuleEntity {
+  key: string; // rule_id
+  rule_name: { top: RuleName[] };
```

Comment:
we might be able to type these using known types, which is preferable but probably not critical. maybe `AggregationsTopMetricsAggregate` would be fitting here.